src/quo2/components/icon.cljs

```diff
@@ -12,33 +12,37 @@
       (and (string? color)
            (not (string/blank? color)))))
 
+(defn- image-icon-style
+  [{:keys [color no-color size container-style theme]}]
+  (cond-> {:width  size
+           :height size}
+    (not no-color)
+    (assoc :tint-color
+           (if (and (string? color) (not (string/blank? color)))
+             color
+             (colors/theme-colors colors/neutral-100 colors/white theme)))
+    :always
+    (merge container-style)))
+
 (defn memo-icon-fn
-  [{:keys [color color-2 no-color
-           container-style size accessibility-label theme]
-    :or   {accessibility-label :icon}}
+  [{:keys [color color-2 container-style size accessibility-label]
+    :or   {accessibility-label :icon}
+    :as   props}
    icon-name]
   (let [size (or size 20)]
-    ^{:key icon-name}
-    (if-let [svg-icon (icons.svg/get-icon icon-name size)]
-      [svg-icon
-       {:size                size
-        :color               (when (valid-color? color) color)
-        :color-2             (when (valid-color? color-2) color-2)
-        :accessibility-label accessibility-label
-        :style               container-style}]
-      [rn/image
-       {:style
-        (merge {:width  size
-                :height size}
-
-               (when (not no-color)
-                 {:tint-color (if (and (string? color) (not (string/blank? color)))
-                                color
-                                (colors/theme-colors colors/neutral-100 colors/white theme))})
-
-               container-style)
-        :accessibility-label accessibility-label
-        :source (icons/icon-source (str (name icon-name) size))}])))
+    (with-meta
+      (if-let [svg-icon (icons.svg/get-icon icon-name size)]
+        [svg-icon
+         (cond-> {:size                size
+                  :accessibility-label accessibility-label
+                  :style               container-style}
+           (valid-color? color)   (assoc :color color)
+           (valid-color? color-2) (assoc :color-2 color-2))]
```

Comment:
Chaged:
```clojure
{:color   (when something? color
 :color-2 (when something-else? color-2
 ,,,}
```

to use `cond->`

The problem here, is that many `svg-icon` components are doing an `:or` destructuring. The `:or` destructuring is applied when the keys don't exists in the map and in the previous code we are adding the keys with a `nil` value, so the destructuring is not applied. Example:
```clojure
(let [{:keys [foo]
       :or   {foo "new value"}} {}]
  foo)
;; => "new value"
(let [{:keys [foo]
       :or   {foo "new value"}} {:foo nil}] ; like the result of a `when` 
  foo)
;; => nil
```